### ISSUE TITLE
Questionnaire profile - unable to resolve name reference #Questionnaire.item

### ISSUE THREAD

**Issue by timster** [standardhealth/shr-cli#221]
I'm not sure where exactly the problem is, but here's what I'm experiencing...

I have a CIMPL file that looks like this:

```
Grammar: Map 6.0
Namespace: del
Target: FHIR_R4

StandardForm maps to Questionnaire:
    constrain name to 1..1
    constrain version to 1..1
    constrain title to 1..1
    constrain description to 1..1
```

Using shr-cli and org.hl7.fhir.publisher.jar, I'm able to build the StructureDefinition and push this out to the server. It can be found here: 

https://impact-fhir.mitre.org/r4/StructureDefinition/del-StandardForm

However, I get an error when I try to load a questionnaire that uses the profile.

The exception raised by the HAPI server is:

`Caused by: org.hl7.fhir.exceptions.DefinitionException: Unable to resolve name reference #Questionnaire.item at path Questionnaire.item.item`

The problem comes when the server tries to validate `"contentReference": "#Questionnaire.item"` under the `"id": "Questionnaire:del-StandardForm.item.item"` section.

It cannot seem to find `#Questionnaire.item`.

I think it should be referencing `#Questionnaire:del-StandardForm.item` instead. It works when I change the profile manually to use `#Questionnaire:del-StandardForm.item` for `contentReference` instead.

Example Questionnaire that fails when POSTing to server.

```
{
  "resourceType": "Questionnaire",
  "meta": {
    "profile": "https://impact-fhir.mitre.org/r4/StructureDefinition/del-StandardForm"
  },
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">Test Questionnaire</div>"
  },
  "status": "draft",
  "name": "ProfileTest",
  "version": "1.0.0",
  "item": [
    {
      "linkId": "Section-1",
      "text": "Section 1",
      "type": "group",
      "readOnly": true,
      "item": [
        {
          "linkId": "Section-1/Question-1",
          "prefix": "Question 1",
          "text": "Example Question 1",
          "type": "text",
          "repeats": false,
          "readOnly": false,
          "maxLength": 2
        }
      ]
    }
  ]
}
```

Should the profile be using `Questionnaire:del-StandardForm.item` instead? Or do I need to do something else to get it to find the reference to `Questionnaire.item`

**Comment by cmoesel** [comment]
@timster -- I think your analysis is correct; it's failing to find the `Questionnaire.item` because the profile actually uses the id `Questionnaire:del-StandardForm.item`.

In previous versions of FHIR, adding the profile identifier to each element id was common practice.  I've just done some poking around and it looks like maybe this is not the case anymore.  I will experiment with _removing_ the profile identifier (e.g., `:del-StandardForm`) from the id.  If it seems to not break anything, we'll go with that and include the fix in our next release.

**Comment by cmoesel** [comment]
Confirmed as fixed in SHR CLI 6.4.0.
